### PR TITLE
[SPARK-41310][CONNECT][PYTHON] Implement DataFrame.toDF

### PR DESCRIPTION
**Pull request by amaliujia**
### What changes were proposed in this pull request?

Implement DataFrame.toDF by reusing existing `RenameColumnsBySameLengthNames` proto.

### Why are the changes needed?

API coverage.

### Does this PR introduce _any_ user-facing change?

NO

### How was this patch tested?

UT

### CODE REVIEW

**Comment by amaliujia** [comment]
@zhengruifeng

File: python/pyspark/sql/connect/dataframe.py [hunk 1]
```diff
@@ -1068,6 +1068,23 @@ def inputFiles(self) -> List[str]:
         query = self._plan.to_proto(self._session.client)
         return self._session.client._analyze(query).input_files
 
+    def toDF(self, *cols: str) -> "DataFrame":
```
Comment (amaliujia) [inline python/pyspark/sql/connect/dataframe.py:1071]: I found hard to support `def toDF(self, *cols: "ColumnOrName") -> "DataFrame":` for the Connect case.

How can we convert a Column to a string? This PySpark API is quite different from Scala side as Scala API only supports a list of strings ` def toDF(colNames: String*): DataFrame`.

**Review by zhengruifeng — APPROVED** [review]

File: python/pyspark/sql/connect/dataframe.py (hunk 1 above)
Comment (zhengruifeng) [reply python/pyspark/sql/connect/dataframe.py:1071]: I suspect the typing `ColumnOrName ` in PySpark is incorrect

Comment (amaliujia) [reply python/pyspark/sql/connect/dataframe.py:1071]: ^ yes

Comment (zhengruifeng) [reply python/pyspark/sql/connect/dataframe.py:1071]: cc @HyukjinKwon

File: python/pyspark/sql/connect/dataframe.py [hunk 2]
```diff
@@ -1068,6 +1068,23 @@ def inputFiles(self) -> List[str]:
...
+            a tuple of string new column name or :class:`Column`. The length of the
+            list needs to be the same as the number of columns in the initial
+            :class:`DataFrame`
+
+        Returns
+        -------
+        :class:`DataFrame`
+            DataFrame with new column names.
+        """
+        return DataFrame.withPlan(plan.RenameColumns(self._plan, list(cols)), self.sparkSession)
```
Comment (zhengruifeng) [inline python/pyspark/sql/connect/dataframe.py:1086]:
```suggestion
        return DataFrame.withPlan(plan.RenameColumns(self._plan, list(cols)), session=self._session)
```

Comment (amaliujia) [reply python/pyspark/sql/connect/dataframe.py:1086]: oh right!

File: python/pyspark/sql/connect/dataframe.py (hunk 1 above)
Comment (HyukjinKwon) [reply python/pyspark/sql/connect/dataframe.py:1142]: Yeah, should fix them.

**Review by HyukjinKwon — APPROVED** [review]

File: python/pyspark/sql/connect/dataframe.py (hunk 1 above)
Comment (HyukjinKwon) [reply python/pyspark/sql/connect/dataframe.py:1142]: https://github.com/apache/spark/pull/38834

**Comment by zhengruifeng** [comment]
merged into master

**Pull request closed without merging**